fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -3303,6 +3303,36 @@ public CommandProcessingResult creditBalanceRefund(Long loanId, JsonCommand comm
 
     }
 
+    @Override
+    @Transactional
+    public CommandProcessingResult markLoanAsFraud(Long loanId, JsonCommand command) {
+        Loan loan = this.loanAssembler.assembleFrom(loanId);
+        final Map<String, Object> changes = new LinkedHashMap<>();
+
+        this.loanEventApiJsonValidator.validateMarkAsFraudLoan(command.json());
```

Comment:
Let's do the validation at the beginning, before assemling the Loan entity. In that case we can save some DB load if there's a validation error.

Reply:
Done